packages/lexical-devtools/src/devtools/index.ts

```diff
@@ -16,46 +16,118 @@ chrome.devtools.panels.create(
   },
 );
 
+chrome.runtime.onConnect.addListener((port) => {
+  port.onMessage.addListener((message) => {
+    if (message.name === 'highlight') {
+      chrome.devtools.inspectedWindow.eval(`
+        highlight('${message.lexicalKey}');
+      `);
+    }
+
+    if (message.name === 'dehighlight') {
+      chrome.devtools.inspectedWindow.eval(`
+        dehighlight();
+      `);
+    }
+  });
+});
+
 function handleShown() {
-  // Use devtools.inspectedWindow.eval() to get editorState updates. For more info on security concerns:
+  // Security concerns related to chrome.devtools.inspectedWindow.eval():
   //   https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Content_scripts
   //   https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/devtools/inspectedWindow/eval
-  chrome.devtools.inspectedWindow
-    // Attach a registerUpdateListener within the window to subscribe to changes in editorState.
-    // After the initial registration, all editorState updates are done via browser.runtime.onConnect & window.postMessage
-    .eval(
-      `
-      const editor = document.querySelectorAll('div[data-lexical-editor]')[0]
-        .__lexicalEditor;
-      
-      const initialEditorState = editor.getEditorState();
-      
-      // custom JSON serialization
-      // existing editorState.toJSON() does not contain Lexical ._key property
-      const serializeEditorState = (editorState) => {
-        const nodeMap = Object.fromEntries(editorState._nodeMap);
-        return {nodeMap};
-      };
-      
+
+  // query document for Lexical editor instance.
+  // TODO: add support multiple Lexical editors within the same page
+  // lexicalKey is attached to each DOM node like so: DOMNode[__lexicalKey_{editorKey}]
+  chrome.devtools.inspectedWindow.eval(`
+    const editorDOMNode = document.querySelectorAll(
+      'div[data-lexical-editor]',
+    )[0];
+    const editor = editorDOMNode.__lexicalEditor;
+    const editorKey = editorDOMNode.__lexicalEditor._key;
+    const lexicalKey = '__lexicalKey_' + editorKey;
+  `);
+
+  // depth first search to find DOM node with lexicalKey
+  chrome.devtools.inspectedWindow.eval(`
+    const findDOMNode = (node, targetKey) => {
+      if (targetKey === 'root') {
+        return editorDOMNode;
+      }
+
+      if (node[lexicalKey] && node[lexicalKey] === targetKey) {
+        return node;
+      }
+    
+      for (let i = 0; i < node.childNodes.length; i++) {
+        const child = findDOMNode(node.childNodes[i], targetKey);
+        if (child) return child;
+      }
+    
+      return null;
+    };
+  `);
```

Comment:
@noi5e Maybe do this in another PR, but if you want to be able to use the eval code outside of the eval statement so you get actual TS errors, just out it in it's own function scope:

```suggestion
  const getEvalCode = () => {
    const findDOMNode = (node, targetKey) => {
      if (targetKey === 'root') {
        return editorDOMNode;
      }

      if (node[lexicalKey] && node[lexicalKey] === targetKey) {
        return node;
      }
    
      for (let i = 0; i < node.childNodes.length; i++) {
        const child = findDOMNode(node.childNodes[i], targetKey);
        if (child) return child;
      }
    
      return null;
    };
  }

  chrome.devtools.inspectedWindow.eval(`${getEvalCode()}`);
```